go/geth_adapter/adapter_test.go

```diff
@@ -75,3 +81,383 @@ func TestRunContextAdapter_SetBalanceHasCorrectEffect(t *testing.T) {
+
+	snapshot := tosca.Snapshot(1)
+
+	stateDb.EXPECT().RevertToSnapshot(int(snapshot))
+	adapter.RestoreSnapshot(snapshot)
+
+	stateDb.EXPECT().Snapshot().Return(int(snapshot))
+	got := adapter.CreateSnapshot()
+	if got != snapshot {
+		t.Errorf("Got wrong snapshot %v, expected %v", got, snapshot)
+	}
+}
+
+func TestRunContextAdapter_AccountOperations(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	stateDb := NewMockStateDb(ctrl)
+	adapter := &runContextAdapter{evm: &geth.EVM{StateDB: stateDb}}
+
+	address := tosca.Address{0x42}
+	code := tosca.Code{1, 2, 3}
+
+	stateDb.EXPECT().AddressInAccessList(gc.Address(address)).Return(false)
+	stateDb.EXPECT().AddAddressToAccessList(gc.Address(address))
+	accessStatus := adapter.AccessAccount(address)
+	if accessStatus != tosca.ColdAccess {
+		t.Errorf("Got wrong access type %v, expected %v", accessStatus, tosca.ColdAccess)
+	}
+
+	stateDb.EXPECT().Exist(gc.Address(address)).Return(true)
+	exits := adapter.AccountExists(address)
+	if !exits {
+		t.Errorf("Account should exist")
+	}
+
+	stateDb.EXPECT().Exist(gc.Address(address)).Return(false)
+	stateDb.EXPECT().CreateAccount(gc.Address(address))
+	stateDb.EXPECT().SetCode(gc.Address(address), code)
+	created := adapter.CreateAccount(address, code)
+	if !created {
+		t.Errorf("Account should have been created")
+	}
+
+	stateDb.EXPECT().AddressInAccessList(gc.Address(address)).Return(true)
+	inAccessList := adapter.IsAddressInAccessList(address)
+	if !inAccessList {
+		t.Errorf("Address should be in access list")
+	}
+}
+
+type testContractRef struct {
+	address gc.Address
+}
+
+func (c testContractRef) Address() gc.Address {
+	return c.address
+}
+
+func TestRunContextAdapter_Call(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	stateDb := NewMockStateDb(ctrl)
+
+	address := gc.Address{0x42}
+
+	stateDb.EXPECT().Snapshot().Return(1)
+	stateDb.EXPECT().Exist(address).Return(true)
+	stateDb.EXPECT().GetCode(address).Return([]byte{})
+	stateDb.EXPECT().Witness()
+
+	contractRef := testContractRef{address: address}
+
+	canTransfer := func(geth.StateDB, gc.Address, *uint256.Int) bool { return true }
+	transfer := func(geth.StateDB, gc.Address, gc.Address, *uint256.Int) {}
+
+	runContextAdapter := &runContextAdapter{
+		evm: &geth.EVM{
+			StateDB: stateDb,
+			Context: geth.BlockContext{
+				CanTransfer: canTransfer,
+				Transfer:    transfer,
+			},
+		},
+		contract: geth.NewContract(contractRef, contractRef, nil, 0),
+	}
+
+	gas := tosca.Gas(42)
+
+	parameters := tosca.CallParameters{
+		Recipient: tosca.Address(address),
+		Gas:       gas,
+	}
+
+	result, err := runContextAdapter.Call(tosca.Call, parameters)
+	if err != nil {
+		t.Errorf("Unexpected error: %v", err)
+	}
+	if !result.Success {
+		t.Errorf("Call was not successful")
+	}
+	if result.GasLeft != gas {
+		t.Errorf("Call has the wrong amount of gas left: %v, expected: %v", result.GasLeft, gas)
+	}
+}
+
+type parametersMatcher struct {
```

Comment:
nit: move to the end of the file, and maybe document why this cannot be [gomock.Eq() ](https://pkg.go.dev/github.com/golang/mock/gomock#Eq)